libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.h

```diff
@@ -20,14 +20,22 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  *
+ * Modified by Jeroen Döll, June 2018
  */
 
 #ifndef ESP8266HTTPClient_H_
 #define ESP8266HTTPClient_H_
 
+#define KEEP_PRESENT_API
+
 #include <memory>
 #include <Arduino.h>
+
+#ifdef KEEP_PRESENT_API
 #include <WiFiClient.h>
+#else
+#include <Client.h>
```

Comment:
Why move to abstract `Client` from `WiFiClient`?  I don't remember ever seeing anything else take this root class.

Reply:
The idea was that this class could be used for other Clients as well, eg an EthernetClient as defined here (https://github.com/esp8266/Arduino/blob/master/libraries/Ethernet/src/EthernetClient.h). But because most people will use the WiFiClient, I will change Client to WiFiClient